File: CS/BeginEndDataUpdateExample/ViewModel/BeginEndUpdateViewModel.cs
using DataUpdatesExample;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BeginEndDataUpdateExample
{
    public class BeginEndUpdateViewModel : UpdateViewModelBase
    {
        public IGridUpdateService GridUpdateService {
            get { return this.GetService<IGridUpdateService>(); }
        }
        protected override void ConfigureSettings() {
            dataSourceSize = 5000;
            groupsCount = dataSourceSize / 30;
            setAllRowValuesOnUpdate = true;
            collectionUpdatesPerSecond = dataSourceSize;
            rowUpdatesPerSecond = dataSourceSize;
        }
        protected override void UpdateData(object sender, EventArgs e) {
            GridUpdateService.BeginDataUpdate();
            UpdateDataCore();
            GridUpdateService.EndDataUpdate();
        }
    }
}

File: VB/BeginEndDataUpdateExample/ViewModel/BeginEndUpdateViewModel.vb
Imports DataUpdatesExample
Imports System

Namespace BeginEndDataUpdateExample

    Public Class BeginEndUpdateViewModel
        Inherits UpdateViewModelBase

        Public ReadOnly Property GridUpdateService As IGridUpdateService
            Get
                Return GetService(Of IGridUpdateService)()
            End Get
        End Property

        Protected Overrides Sub ConfigureSettings()
            dataSourceSize = 5000
            groupsCount = dataSourceSize \ 30
            setAllRowValuesOnUpdate = True
            collectionUpdatesPerSecond = dataSourceSize
            rowUpdatesPerSecond = dataSourceSize
        End Sub

        Protected Overrides Sub UpdateData(ByVal sender As Object, ByVal e As EventArgs)
            GridUpdateService.BeginDataUpdate()
            UpdateDataCore()
            GridUpdateService.EndDataUpdate()
        End Sub
    End Class
End Namespace
